src/main/java/AddNullCheckBeforeDereferenceRefactoring.java

```diff
@@ -49,13 +40,16 @@ public boolean isApplicable(ASTNode node) {
 		if (node instanceof IfStatement ifStmt) {
 			Expression condition = ifStmt.getExpression();
 			if (condition instanceof InfixExpression infix
-					&& infix.getOperator() == InfixExpression.Operator.NOT_EQUALS) {
+					&& (infix.getOperator() == InfixExpression.Operator.NOT_EQUALS
+							|| infix.getOperator() == InfixExpression.Operator.EQUALS)) {
 				Expression leftOperand = infix.getLeftOperand();
 				Expression rightOperand = infix.getRightOperand();
 
 				if ((leftOperand instanceof SimpleName && rightOperand instanceof NullLiteral)
-						|| (rightOperand instanceof SimpleName && leftOperand instanceof NullLiteral)) {
-					System.out.println("[DEBUG] Found indirect null check in if-statement: " + condition);
+						|| (rightOperand instanceof SimpleName
+								&& leftOperand instanceof NullLiteral)) {
+					System.out.println("[DEBUG] Found indirect null check in if-statement: "
+							+ condition);
```

Comment:
I don't think there are actually any changes at this line, only formatting. Please avoid unnecessary diffs - the reviewer's attention should be directed only to things that actually changed and that actually matter. The easiest way to avoid this problem is to use an autoformatter like Spotless (which should be applied in a separate PR, so that no one has to look carefully at the huge amounts of formatting changes it introduces).